### PR TITLE
Nix Flakes and IPFS integration

### PR DESCRIPTION
**Pull request by mpontus**
- Yarn updated to v3.
- Paths are prefixed
- Fonts are inlined
- Careers are loaded in runtime
- New commands: `nix build`, `nix develop`
- New action: [deploy-ipfs.yml](https://github.com/kfactory-dev/kf-website/blob/flakes/.github/workflows/deploy-ipfs.yml)
- [Readme](https://github.com/kfactory-dev/kf-website/tree/flakes#readme) updated

### CODE REVIEW

File: .github/workflows/deploy-ipfs.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+on: push
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: install Nix
+        uses: cachix/install-nix-action@v15
```
Comment (asymmetric) [inline .github/workflows/deploy-ipfs.yml:11]: v16 is out btw

File: default.nix [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# This is a minimal `default.nix` by yarn-plugin-nixify. You can customize it
+# as needed, it will not be overwritten by the plugin.
+
+{ pkgs ? import <nixpkgs> { } }:
+
+pkgs.callPackage ./yarn-project.nix { } { src = ./.; }
```
Comment (asymmetric) [inline default.nix:6]: If you're only gonna support nix flakes, then you don't need this file. There's actually also a config setting for the plugin, `generateDefaultNix`, which would not generate it.

File: flake.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  };
+
+  outputs = { self, nixpkgs, flake-utils }:
+    flake-utils.lib.eachDefaultSystem (system:
+      with import nixpkgs { inherit system; };
+      let project = pkgs.callPackage ./yarn-project.nix { } { src = ./.; };
+      in {
+
+        defaultPackage = project.overrideAttrs (oldAttrs: {
+          buildInputs = oldAttrs.buildInputs ++ [ pkgs.python3 ];
```
Comment (asymmetric) [inline flake.nix:16]: Do you need actually need python here?

File: flake.nix [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+          buildPhase = "yarn build";
+          installPhase = "mv public $out";
+        });
```
Comment (asymmetric) [inline flake.nix:19]: I think you might not need to do any of this. I think `project` is already your built app.

Comment (asymmetric) [reply flake.nix:19]: But I haven't tested this!

File: flake.nix (hunk 3 above)
Comment (mpontus) [reply flake.nix:16]: Yes, python3 is needed to build node-gyp modules (`lmdb-store` and `msgpackr-extract`)

File: flake.nix (hunk 4 above)
Comment (mpontus) [reply flake.nix:19]: yarn-plugin-nixify does not appear to create `defauilt.nix` when `flake.nix` is present in the same directory, and then it also differs from what you see above: here we execute build step and copy only the static files to the output directory

**Pull request merged**